static/src/javascripts/projects/common/modules/experiments/ab.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+import type { ABTest } from 'common/modules/experiments/ab-types';
+
+import {
+    getActiveTests,
+    getTest,
+    TESTS,
+} from 'common/modules/experiments/ab-tests';
+import { buildOphanSubmitter } from 'common/modules/experiments/ab-ophan';
+import * as segmentUtil from 'common/modules/experiments/segment-util';
+import * as testCanRunChecks
+    from 'common/modules/experiments/test-can-run-checks';
+import * as abUtils from 'common/modules/experiments/utils';
+import { local } from 'lib/storage';
+
+const noop = (): null => null;
+
+// Finds variant in specific tests and runs it
+const runTest = test => {
+    if (abUtils.isParticipating(test) && testCanRunChecks.testCanBeRun(test)) {
+        const participations = abUtils.getParticipations();
+        const variantId = participations[test.id].variant;
+        const variant = abUtils.getVariant(test, variantId);
+
+        if (variant) {
+            variant.test();
+        } else if (!segmentUtil.isInTest(test) && test.notInTest) {
+            test.notInTest();
+        }
+    }
+};
+
+const allocateUserToTest = test => {
+    // Only allocate the user if the test is valid and they're not already participating.
+    if (testCanRunChecks.testCanBeRun(test) && !abUtils.isParticipating(test)) {
+        abUtils.addParticipation(test, segmentUtil.variantIdFor(test));
+    }
+};
+
+const getForcedIntoTests = () => {
+    if (/^#ab/.test(window.location.hash)) {
+        const tokens = window.location.hash.replace('#ab-', '').split(',');
+
+        return tokens.map(token => {
+            const abParam = token.split('=');
+
+            return {
+                id: abParam[0],
+                variant: abParam[1],
+            };
+        });
+    }
+
+    return JSON.parse(local.get('gu.devtools.ab')) || [];
+};
+
+export const shouldRunTest = (id: string, variant: string) => {
+    const test = getTest(id);
+
+    return (
+        test &&
+        abUtils.isParticipating(test) &&
+        abUtils.getTestVariantId(id) === variant &&
+        testCanRunChecks.testCanBeRun(test)
+    );
+};
+
+export const segment = (tests: Array<ABTest>) =>
+    tests.forEach(allocateUserToTest);
+
+export const forceSegment = (testId: string, variant: string) => {
+    getActiveTests().filter(test => test.id === testId).forEach(test => {
```

Comment:
If `testId` is unique then we can probably switch to `let test = getActiveTests().find(test => test.id = testId)` instead?